Assets/Scripts/PlayerController.cs

```diff
@@ -36,42 +39,60 @@ void Start()
 
     void Update()
     {
-        // --------------------------------------
-        // -----------Movement Controls----------
-        // --------------------------------------
-        float moveHorizontal = Input.GetAxis("Horizontal");
-        float moveVertical = Input.GetAxis("Vertical");
-
         // Calculate the camera's forward and right directions
         Vector3 cameraForward = mainCamera.transform.forward;
         cameraForward.y = 0.0f;
         cameraForward.Normalize();
         Vector3 cameraRight = mainCamera.transform.right;
 
-        // Handle player movement
+        // Get player movement
         Vector3 movement = new Vector3(
             Input.GetAxis("Horizontal"),
             0.0f,
             Input.GetAxis("Vertical")
         );
-        Vector3 movementDirection = cameraForward * movement.z + cameraRight * movement.x;
-        movementDirection.y = 0.0f; // Prevent player from moving up/down
-        movementDirection.Normalize();
-        transform.position += movementDirection * moveSpeed * Time.deltaTime;
-
-        // ------------------------------------
-        // ---------Animation Controls---------
-        // ------------------------------------
-        if (movementDirection != Vector3.zero)
+            
+        if(!inputDisabled){
```

Comment:
This isn't required yet, but it will  be useful when we add pause menus and that sort